packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -198,6 +198,142 @@ class RenderEditable extends RenderBox {
 
   Rect _lastCaretRect;
 
+  bool _rightArrowDown = false;
+  bool _leftArrowDown = false;
+  bool _upArrowDown = false;
+  bool _downArrowDown = false;
```

Comment:
In case a user focuses when a key is already held down. Do I not need to worry about this case?

Reply:
How does it help with that case? Can you elaborate?
Also that would be a good test to add.

Reply:
Ok, yah I think you're right actually. Because we get a function call every frame for the button that's been pressed most recently. And you can only really move the cursor in one direction at once (at least on typical keyboards). So I'll make them locals then...